packages/react-tabs/src/utilities/useOverflow.ts

```diff
@@ -75,6 +75,7 @@ export const useOverflow = ({ onOverflowItemsChanged, rtl, pinnedIndex }: Overfl
     return () => containerRef(null);
   });
 
+  // eslint-disable-next-line no-restricted-properties
```

Comment:
@behowell FYI, might be good to look into whether this could safely be updated to use `useIsomorphicLayoutEffect`, or how to handle SSR properly if that's not possible. (no need for changes in this PR though)

Reply:
I'm not familiar enough with SSR to know offhand. Basically this needs to happen at some point--it's ok if it's delayed on the first render after SSR, but it would be bad if it never happened. Is that what `useIsomorphicLayoutEffect` achieves?

Reply:
`useIsomorphicLayoutEffect` does `useLayoutEffect` normally, and `useEffect` if server-rendered. For most things that's okay, but depending on what it's being used for, could cause issues like flashing on first render (or maybe bugs in a few cases that are particularly dependent on the timing).

I'm not sure what a good way is to validate SSR. I'm not sure how to manually test it (would have to look that up). For v8 we have an ssr-tests package, but it's limited to extremely basic verification that the components render.

Reply:
Things that are related to DOM calculations/operations probably should never executed on server at all as there is no DOM 🙄

It depends on use case, but sometimes we can simply exclude `useLayoutEffect` based on environment, for example, it's how it's done in Emotion:
https://github.com/emotion-js/emotion/blob/6c4a9e50f177900f54f7b3368a55b1a7d5e3c002/packages/react/src/global.js#L83-L86